frontend/src/utils/getDeadLineText.ts

```diff
@@ -0,0 +1,19 @@
+const getDeadlineText = (
+  recruitmentStart: Date | null,
+  recruitmentEnd: Date | null,
+  today: Date = new Date(),
+): string => {
+  if (
+    recruitmentStart &&
+    recruitmentEnd &&
+    today >= recruitmentStart &&
+    today <= recruitmentEnd
+  ) {
+    const diffTime = recruitmentEnd.getTime() - today.getTime();
+    const days = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+    return days > 0 ? `D-${days}` : 'D-Day';
+  }
+  return '모집 마감';
+};
```

Comment:
저희가 date-fns 라이브러리를 달력 컴포넌트 제작을 위해 이미 설치해둔 상태라  
날짜 관련 유틸 함수들을 간편하게 사용할 수 있어요!

아래처럼 import해서 적용하면 한두 줄로 가독성 좋게 작성할 수 있고,  
D-n 계산 시 시간 단위에서 발생할 수 있는 오차도 방지할 수 있어서 더 안정적입니다 🙌

```suggestion
import { isAfter, isBefore, isWithinInterval, differenceInCalendarDays } from 'date-fns';
...
  if (!recruitmentStart || !recruitmentEnd) return '모집 마감';
  if (isBefore(today, recruitmentStart)) return '모집 전';
  if (isAfter(today, recruitmentEnd)) return '모집 마감';

  const days = differenceInCalendarDays(recruitmentEnd, today);
  return days > 0 ? `D-${days}` : 'D-Day';
};
```

Reply:
유틸함수 완전 많이 들어있어용

### [date-fns 공식문서](https://date-fns.org/v4.1.0/docs/differenceInCalendarDays)

Reply:
[7d08ab5](https://github.com/Moadong/moadong/pull/312/commits/7d08ab59bd97b6f3b3341915fe071e425a9f05b2) 

이렇게 리팩토링했습니다. 내장 메서드가 있는지 몰랐네요 ㅎㅎ 사용하기 훨씬 편하고 간결해진 것 같습니다.
 ```모집 전```을 추가했고 테스트케이스 또한 추가했습니다~!